Explorer/Assets/Scripts/ECS/Unity/Transforms/Components/TransformComponent.cs

```diff
@@ -0,0 +1,27 @@
+using ECS.ComponentsPooling;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ECS.Unity.Transforms.Components
+{
+    /// <summary>
+    ///     A wrapper over the transform unity component that is created from the SDKTransform component.
+    /// </summary>
+    public class TransformComponent : IPoolableComponentProvider
+    {
+        public Transform Transform;
+        public HashSet<TransformComponent> Children;
+        public TransformComponent Parent;
```

Comment:
I see why you have the necessity to reference `TransformComponent` as a class but I think:
1. It's error-prone and obscure
2. It's against ECS rules: you can't cross-reference components, you should act as if all of them are value types and can't be referenced directly

Thus, I propose changing it to `Entity` (check `EntityReference`) so you will need to query it from the world. Also considering you introduced an entity for the scene root, that entity can be also stored there.